### DISCUSSION TITLE
Sharing Data from JavaScript to Bevy via WASM

### DISCUSSION THREAD

**Discussion by vinnyhoward** [bevyengine/bevy#8861] [Q&A]
I'm creating a Bevy game in the browser with WASM and need to pass user inputs (in this case a URL string) from JavaScript to a Bevy system. I've tried using `Arc<Mutex<_>>` to share state across threads and have tried several other approaches but I'm still struggling to update a shared resource within the Bevy system from the WASM function.

How can I effectively share JavaScript user inputs with a Bevy system via WASM so I can dynamically alter the game based on these inputs?

EDIT: For more context, I have a menu state and when the user loads their crypto wallet along with their NFT's which has the asset URLs. Once they hit start I want to go into a loading state. In this loading state is where I would need to grab the user's assets that were saved via WASM/JS

**Comment by bennobuilder** [comment]
I'm also exploring options to share Events / Data between Javascript and Bevy. I also found myself using a `Mutex`. But I'm wondering what the best practices are: https://github.com/bevyengine/bevy/discussions/10172 Did you made any progress figuring out a good solution? Thanks :)
